### DISCUSSION TITLE
Give me some way to read back a color

### DISCUSSION THREAD

**Discussion by mcclure** [PintaProject/Pinta#646] [Ideas]
I have an image. There's a particular pixel, and I want to know what its color is so I can type it into another program. I open the program in Pinta, I select the eyedropper, I select the appropriate pixel, and there's my color! … now what?

There's no way, having put a color into the eyedropper, to read back its RGBA value. Double clicking the color does nothing. Making the color primary/secondary then double clicking it takes you to a color selector, but this only lets you dial in new colors, you can't (even though this would be useful) bring an existing color into the (+) color selector. There's an eyedropper in the "choose primary color" dialog that appears to do nothing.

I think there should be *some way* to ask Pinta what a color is. This could be any of: Displaying it in the bottom info bar; displaying it in the top Tool: bar when using the eyedropper; showing it when a color is selected in the "choose primary color" dialog; or making it possible to open one of the "Custom" colors from the "Choose primary color" dialog in the color picker.

EDIT: Note, I am using Pinta 2.1.1 from the Ubuntu Snap Store.

**Comment by cameronwhite** [comment]
I agree the standard GTK color chooser dialog isn't great (see also #570), and has a lot of hidden / hard to discover functionality

When you double click the primary / secondary color to open the dialog, the color should show up under the "Custom" section at the bottom and be selected. If you then right click on it and select "Customize", then this brings the color into the edit window where you can get its hex code value

**Reply by mcclure** [reply]
Thank you, that is very helpful.

I wonder if there is a more direct/convenient way to foreground this in the UI that a future version of Pinta could introduce, but this unblocks me for now.

**Reply by cameronwhite** [reply]
With the new color picker dialog in Pinta 3.0 it's much easier to get the color values from the current color or any of the palette colours, e.g. the video below


https://github.com/user-attachments/assets/8739f3cc-44d6-424a-8136-f74e901c762c

**Comment by PeCoBe** [comment]
Paint 1, Pinta 0.
It should be more visible to give a better user experience.
